Connect volunteer form to back end

JohnathonBowers opened the pull request:
## Description

These changes update the `onSubmit` function in the Volunteer form to make a POST request to the backend API with the volunteer form data.

## Issues

Resolves Issue #24 

## Screenshots

The screen recording below shows the form data successfully posting to the Volunteer Form tab on the back end Google spreadsheet:

https://github.com/allie-rae/rescue-the-birds/assets/109883576/82e5f742-ffce-4f3c-8c5e-d2d6d694437e

And this screen recording shows the error response if the form submission fails. (To trigger this error, I entered an invalid URL for the HTTP POST request.)

https://github.com/allie-rae/rescue-the-birds/assets/109883576/2da1b8e5-4db3-4127-abf9-4986074bc56e

JohnathonBowers committed:
Add initial code for submitting form to backend

Signed-off-by: Johnathon Bowers <johnathonbowers@gmail.com>

allie-rae reviewed (changes requested):
This all looks great to me, except one odd thing. Some of your input labels aren't snapping back into place post-submission. 🤔 I'm not sure why it's happening, but I'll leave the investigation to you! It's only the last 5 inputs. The others are behaving as expected. Here's a screenshot: 
<img width="633" alt="Screenshot 2024-01-05 at 5 35 23 PM" src="https://github.com/allie-rae/rescue-the-birds/assets/48700332/e5d9e870-9a5e-444c-9e8d-c793f37de4b4">

JohnathonBowers committed:
Wrap TextField component in Controller component to fix data-shrink prop issue

Signed-off-by: Johnathon Bowers <johnathonbowers@gmail.com>

JohnathonBowers committed:
Add general validation error message to bottom of form

Signed-off-by: Johnathon Bowers <johnathonbowers@gmail.com>

JohnathonBowers commented:
I fixed the problem, @allie-rae ! Thank you for catching it!

First, the symptom: The `data-shrink` prop in the `MuiInputLabelRoot` component (which is rendered a little bit farther up the tree by the `TextField` component) was staying `true` even after the form was submitted and the `reset()` function was invoked.

The diagnosis: Uncontrolled-TextField-Component-itis

The prescription: React Hook Form's `Controller` component

By wrapping each of the TextField components in React Hook Form's `Controller` component, MaterialUI's components get the updated state information they need to flip the `data-shrink` prop back to `false`. Wahoo!

Here's a screen recording of our healthy patient:

https://github.com/allie-rae/rescue-the-birds/assets/109883576/f5d591fd-c33b-4d9e-a14e-3cc0d0dc79ef


And here's a screenshot of the error response if the form fails to submit:

<img width="1666" alt="Screenshot 2024-01-06 at 11 49 07 PM" src="https://github.com/allie-rae/rescue-the-birds/assets/109883576/dc266a22-3046-49ed-a0ca-e0ef90721f71">


One last thing: I added a generic validation error message below the "Submit" button to let the user know if they missed a required field in the form. Granted, they'll be alerted in real time if they tab through a required field, but I could envision a scenario in which someone accidentally skipped one. Because of the blocks of text at the bottom of the form, none of the field-specific validation error messages show up in the viewport, which could leave a user in the unenviable position of clicking on an unresponsive "Submit" button. This generic error message can tip them off that something is missing:

<img width="1666" alt="Screenshot 2024-01-06 at 11 50 31 PM" src="https://github.com/allie-rae/rescue-the-birds/assets/109883576/6292bd0d-0a04-4d53-94fb-714a0f3d2b30">

allie-rae reviewed (approved):
Great job!! I love the solution you came up with and the addition of the "Please fill out all required fields" error messaging. MVP done!!
